heron/spi/src/java/com/twitter/heron/spi/statemgr/IStateManager.java

```diff
@@ -276,4 +277,27 @@ public String getName() {
    */
   ListenableFuture<PhysicalPlans.PhysicalPlan> getPhysicalPlan(
       WatchCallback watcher, String topologyName);
+
+  /**
+   * Set the Stateful Checkpoint
+   *
+   * @return Boolean - Success or Failure
+   */
+  ListenableFuture<Boolean> setStatefulCheckpoint(
+      CheckpointManager.StatefulConsistentCheckpoints checkpoint, String topologyName);
+
+  /**
+   * Get the Stateful Checkpoint
+   *
+   * @return Boolean - Success or Failure
+   */
+  ListenableFuture<CheckpointManager.StatefulConsistentCheckpoints> getStatefulCheckpoint(
```

Comment:
The object contains multiple checkpoints so we should put an 's' on the method name. Also can it take some identifier of the currently topology like the packing plan id? For example upon restarting a topology we need to know if the previous checkpoints are still valid for the new plan, if it's been changed between deploys.